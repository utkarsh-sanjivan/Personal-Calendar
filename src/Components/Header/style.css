.calendar-header-container {
    padding-top: 30px;    
    margin-bottom: 20px;
    display: flex;
}

.header-text {
    margin: 0px 10px;
    min-width: 70px;
    width: 70px;
}

.today-button {
    border-radius: 30px;
}

.nav-button {
    border: none;
    box-shadow: none;
}

.search-input {
    width: 100%;
    border-radius: 20px;
}