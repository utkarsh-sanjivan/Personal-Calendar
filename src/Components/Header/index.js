import React from 'react';
import { Select } from 'antd';
import CalendarButton from '../CalendarButton/index';
import Search from './../Search/index';
import { LeftOutlined, RightOutlined } from '@ant-design/icons';
import { getMonthString } from './../../Utils/dateFromatter';
import { MONTHS } from './../../Utils/constant';
import './style.css';
const { Option } = Select;

const Header = props => (
    <div className="calendar-header-container">
        <div className='calendar-header-left'>
            <div>
                <CalendarButton className="today-button" onClick={props.getToday}>Today</CalendarButton>
                <span>
                    <CalendarButton className="nav-button" onClick={() => props.navMonth('prev')}><LeftOutlined /></CalendarButton>
                    <CalendarButton className="nav-button" onClick={() => props.navMonth('next')}><RightOutlined /></CalendarButton>
                </span>
            </div>
            <div  className='calendar-header-top-right'>
                <Select 
                    className="search-month-year-input" 
                    defaultValue={getMonthString(props.currentDate.getMonth())} 
                    onChange={props.onChangeMonth}
                >
                    {MONTHS.map(month => <Option {...month} value={month.name}>{month.name}</Option>)}
                </Select>
                <Select 
                    className="search-month-year-input" 
                    defaultValue={props.currentDate.getFullYear()} 
                    onChange={props.onChangeYear}
                >
                    {Array(130).fill(null).map((__, index) => <Option index={index} value={index+1970}>{index+1970}</Option>)}
                </Select>
            </div>
        </div>
        <div className='calendar-header-right'>
            <Search 
                className="search-input"
                onSearch={props.onSearch}
                onSelect={props.onSelect}
                autoCompleteData={props.autoCompleteData}
            />
        </div>
    </div>
);

export default Header;