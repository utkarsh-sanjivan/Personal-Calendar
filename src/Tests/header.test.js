import React from 'react';
import { create } from "react-test-renderer";
import Header from './../Components/Header/index';

test('Test Header Component', () => {
const component = create(<Header 
    currentDate={new Date}
    navMonth={() => {}}
    onChangeMonth={() => {}}
    onChangeYear={() => {}}
    getToday={() => {}}
    onSearch={() => {}}
    onSelect={() => {}}
    autoCompleteData={[]}
  />);
  expect(component.toJSON()).toMatchSnapshot()
});