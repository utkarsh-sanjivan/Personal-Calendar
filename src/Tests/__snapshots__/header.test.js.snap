// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Header Component 1`] = `
<div
  className="calendar-header-container"
>
  <div
    className="calendar-header-left"
  >
    <div>
      <button
        className="ant-btn today-button"
        onClick={[Function]}
        type="button"
      >
        <span>
          Today
        </span>
      </button>
      <span>
        <button
          className="ant-btn nav-button"
          onClick={[Function]}
          type="button"
        >
          <span
            aria-label="left"
            className="anticon anticon-left"
            role="img"
          >
            <svg
              aria-hidden="true"
              className=""
              data-icon="left"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
              />
            </svg>
          </span>
        </button>
        <button
          className="ant-btn nav-button"
          onClick={[Function]}
          type="button"
        >
          <span
            aria-label="right"
            className="anticon anticon-right"
            role="img"
          >
            <svg
              aria-hidden="true"
              className=""
              data-icon="right"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
              />
            </svg>
          </span>
        </button>
      </span>
    </div>
    <div
      className="calendar-header-top-right"
    >
      <div
        className="ant-select search-month-year-input ant-select-single ant-select-show-arrow"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
      >
        <div
          className="ant-select-selector"
          onMouseDown={[Function]}
        >
          <span
            className="ant-select-selection-search"
          >
            <input
              aria-activedescendant="undefined_list_0"
              aria-autocomplete="list"
              aria-controls="undefined_list"
              aria-haspopup="listbox"
              aria-owns="undefined_list"
              autoComplete="off"
              className="ant-select-selection-search-input"
              onChange={[Function]}
              onCompositionEnd={[Function]}
              onCompositionStart={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onPaste={[Function]}
              readOnly={true}
              role="combobox"
              style={
                Object {
                  "opacity": 0,
                }
              }
              unselectable="on"
              value=""
            />
          </span>
          <span
            className="ant-select-selection-item"
            title="August"
          >
            August
          </span>
        </div>
        <span
          aria-hidden={true}
          className="ant-select-arrow"
          onMouseDown={[Function]}
          style={
            Object {
              "WebkitUserSelect": "none",
              "userSelect": "none",
            }
          }
          unselectable="on"
        >
          <span
            aria-label="down"
            className="anticon anticon-down ant-select-suffix"
            role="img"
          >
            <svg
              aria-hidden="true"
              className=""
              data-icon="down"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
              />
            </svg>
          </span>
        </span>
      </div>
      <div
        className="ant-select search-month-year-input ant-select-single ant-select-show-arrow"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
      >
        <div
          className="ant-select-selector"
          onMouseDown={[Function]}
        >
          <span
            className="ant-select-selection-search"
          >
            <input
              aria-activedescendant="undefined_list_0"
              aria-autocomplete="list"
              aria-controls="undefined_list"
              aria-haspopup="listbox"
              aria-owns="undefined_list"
              autoComplete="off"
              className="ant-select-selection-search-input"
              onChange={[Function]}
              onCompositionEnd={[Function]}
              onCompositionStart={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onPaste={[Function]}
              readOnly={true}
              role="combobox"
              style={
                Object {
                  "opacity": 0,
                }
              }
              unselectable="on"
              value=""
            />
          </span>
          <span
            className="ant-select-selection-item"
            title="2020"
          >
            2020
          </span>
        </div>
        <span
          aria-hidden={true}
          className="ant-select-arrow"
          onMouseDown={[Function]}
          style={
            Object {
              "WebkitUserSelect": "none",
              "userSelect": "none",
            }
          }
          unselectable="on"
        >
          <span
            aria-label="down"
            className="anticon anticon-down ant-select-suffix"
            role="img"
          >
            <svg
              aria-hidden="true"
              className=""
              data-icon="down"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
              />
            </svg>
          </span>
        </span>
      </div>
    </div>
  </div>
  <div
    className="calendar-header-right"
  >
    <div
      autoCompleteData={Array []}
      className="ant-select search-input ant-select-auto-complete ant-select-single ant-select-customize-input ant-select-show-search"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
    >
      <div
        className="ant-select-selector"
        onMouseDown={[Function]}
      >
        <span
          className="ant-select-selection-search"
        >
          <span
            className="ant-select-selection-search-input ant-input-affix-wrapper"
            onMouseUp={[Function]}
            style={
              Object {
                "opacity": null,
              }
            }
          >
            <span
              className="ant-input-prefix"
            >
              <span
                aria-label="search"
                className="anticon anticon-search search-icon"
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                role="img"
              >
                <svg
                  aria-hidden="true"
                  className=""
                  data-icon="search"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                  />
                </svg>
              </span>
            </span>
            <input
              aria-activedescendant="undefined_list_0"
              aria-autocomplete="list"
              aria-controls="undefined_list"
              aria-haspopup="listbox"
              aria-owns="undefined_list"
              autoComplete="off"
              autoFocus={true}
              className="ant-input"
              onBlur={[Function]}
              onChange={[Function]}
              onCompositionEnd={[Function]}
              onCompositionStart={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onPaste={[Function]}
              placeholder="Search Events"
              readOnly={false}
              role="combobox"
              style={null}
              type="text"
              unselectable={null}
              value=""
            />
            <span
              className="ant-input-suffix"
            >
              <span
                aria-label="calendar"
                className="anticon anticon-calendar"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  className=""
                  data-icon="calendar"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
                  />
                </svg>
              </span>
            </span>
          </span>
        </span>
        <span
          className="ant-select-selection-placeholder"
        />
      </div>
    </div>
  </div>
</div>
`;
